httpserver/server_certificates.h

```diff
@@ -0,0 +1,254 @@
+
+    add_ext(x, NID_subject_key_identifier, (char*)"hash");
+
+    /* Some Netscape specific extensions */
+    //    add_ext(x, NID_netscape_cert_type, "sslCA");
+
+    //    add_ext(x, NID_netscape_comment, "example comment extension");
+
+#ifdef CUSTOM_EXT
+    /* Maybe even add our own extension based on existing */
+    {
+        int nid;
+        nid = OBJ_create("1.2.3.4", "MyAlias", "My Test Alias Extension");
+        X509V3_EXT_add_alias(nid, NID_netscape_comment);
+        add_ext(x, nid, "example comment alias");
+    }
+#endif
+
+    if (!X509_sign(x, pk, EVP_sha256()))
+        goto err;
+
+    *x509p = x;
+    *pkeyp = pk;
+    return (1);
+err:
+    return (0);
+}
+
+void generate_cert(const char* cert_path, const char* key_path) {
+    BIO* bio_err;
+    X509* x509 = NULL;
+    EVP_PKEY* pkey = NULL;
+
+    OpenSSL_add_all_digests();
+
+    CRYPTO_mem_ctrl(CRYPTO_MEM_CHECK_ON);
+
+    bio_err = BIO_new_fp(stderr, BIO_NOCLOSE);
+
+    mkcert(&x509, &pkey, 2048, 1, 10000);
+
+    //    RSA* rsa = EVP_PKEY_get1_RSA(pkey);
+    //    RSA_print_fp(stdout,rsa,0);
+    X509_print_fp(stdout, x509);
+
+    FILE* key_f = fopen(key_path, "wt");
+    if (!PEM_write_PrivateKey(key_f, pkey, NULL, NULL, 0, NULL, NULL))
+        abort();
+    fclose(key_f);
+    FILE* cert_f = fopen(cert_path, "wt");
+    PEM_write_X509(cert_f, x509);
+    fclose(cert_f);
+
+    X509_free(x509);
+    EVP_PKEY_free(pkey);
+
+    CRYPTO_cleanup_all_ex_data();
+
+    //    CRYPTO_mem_leaks(bio_err);
+    BIO_free(bio_err);
+}
+
+inline void load_server_certificate(boost::asio::ssl::context& ctx) {
+    /*
+        The certificate was generated from CMD.EXE on Windows 10 using:
+
+        winpty openssl dhparam -out dh.pem 2048
+        winpty openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days
+       10000 -out cert.pem -subj "//C=US\ST=CA\L=Los
+       Angeles\O=Beast\CN=www.example.com"
+    */
+    const auto cert_path = "./cert.pem";
+    const auto key_path = "./key.pem";
+    const auto dh_path = "./dh.pem";
+    if (!boost::filesystem::exists(cert_path) ||
+        !boost::filesystem::exists(key_path)) {
+        generate_cert(cert_path, key_path);
+    }
+    if (!boost::filesystem::exists(dh_path)) {
+        generate_dh_pem(dh_path);
+    }
+
+    std::ifstream file(cert_path);
+    const std::string cert((std::istreambuf_iterator<char>(file)),
+                           std::istreambuf_iterator<char>());
+    file.close();
+
+    file.open(key_path);
+    const std::string key((std::istreambuf_iterator<char>(file)),
+                          std::istreambuf_iterator<char>());
+    file.close();
+
+    file.open(dh_path);
+    const std::string dh((std::istreambuf_iterator<char>(file)),
+                         std::istreambuf_iterator<char>());
+    file.close();
+
+    ctx.set_password_callback(
+        [](std::size_t, boost::asio::ssl::context_base::password_purpose) {
+            return "test";
```

Comment:
Is this OK?

Reply:
Not used (the keys are not protected) so removed